ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,94 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
             sub_faces.extend(face.sub_faces_by_ratio(ratio))
         return sub_faces
 
+    def _sub_faces_by_dimensions(self, sub_face_height, sub_face_width, wall_length,
+                                dist_breakup, sill_height, rect_height, changed_var):
+        """Get a list of faces with a combined area equal to the glazing ratio
+        times this face area. The glazing ratio is derived internally using the
+        specified sub face height and width along with other parameters.
+
+        All sub faces will lie inside the boundaries of this face.
+
+        Args:
+            height: sub face height
+            width: sub face width
+            wall_length: total length (width) of sub face wall
+            dist_breakup: distance by which main rectangle is meant to be 
+                divided up into points on the bottom
+            sill_height: A number for the target height above the bottom edge of
+                the rectangle to start the sub-rectangles. Note that, if the
+                ratio is too large for the height, the ratio will take precedence
+                and the sub-rectangle height will be smaller than this value.
+            rect_height: A number for the target height of the output rectangles. Note that, if the ratio is too large for the height,
+                the ratio will take precedence and the sub-rectangle height will
+                be larger than this value.
+            changed_var: string indicating the variable being changed 
+
+        Returns:
+            A list of Face3D objects for sub faces.
+        """
+        sub_face_height_final = rect_height if sub_face_height > rect_height else sub_face_height
+        sill_height_final = 0 if sill_height < 0.01 * rect_height else rect_height-0.1 if sill_height > rect_height-0.1 else sill_height
+
+        if sub_face_height_final + sill_height_final > rect_height:
+            if changed_var == "sill_height_value":
+                sub_face_height_final = rect_height - sill_height_final
+            else:
+                sill_height_final = rect_height - sub_face_height_final
+
+        max_width_break_up = wall_length/2
+        num_divisions = round(wall_length/dist_breakup) if wall_length > dist_breakup/2 else 1
+        if sub_face_width < max_width_break_up and num_divisions * sub_face_width <= wall_length and num_divisions != 1:
```

Comment:
Some in-line comments would be extremely helpful here to understand what you are going to do when these criteria are True.